caproto/commandline/cli_print_formats.py

```diff
@@ -0,0 +1,248 @@
+        df.format = "s"
+
+    # 'data' contains an array of floats
+    if(isinstance(data[0], float)):
+        # Check if any of the format specification arguments were passed
+        if args.float_e is not None:
+            df.format = ".{}e".format(args.float_e)
+        elif args.float_f is not None:
+            df.format = ".{}f".format(args.float_f)
+        elif args.float_g is not None:
+            df.format = ".{}g".format(args.float_g)
+        elif args.float_s:
+            # This feature is not implemented yet. Instead use floating point
+            #    value supplied by the server and print it in %f format.
+            #    This is still gives some elementary support for the argument -s.
+            df.format = "f"
+            df.float_server_precision = True
+        elif args.float_lx:  # Rounded hexadecimal
+            df.format = "X"
+            df.prefix = "0x"
+            df.float_round = True
+        elif args.float_lo:  # Rounded octal
+            df.format = "o"
+            df.prefix = "0o"
+            df.float_round = True
+        elif args.float_lb:  # Rounded binary
+            df.format = "b"
+            df.prefix = "0b"
+            df.float_round = True
+
+    # 'data' contains an array of integers
+    if(isinstance(data[0], numbers.Integral)):
+        if args.int_0x:
+            df.format = "X"   # Hexadecimal
+            df.prefix = "0x"
+        elif args.int_0o:
+            df.format = "o"   # Octal
+            df.prefix = "0o"
+        elif args.int_0b:
+            df.format = "b"   # Binary
+            df.prefix = "0b"
+
+    # Separator: may be a single character (quoted or not quoted) or quoted multiple characters
+    #          including spaces. EPICS caget allows only single character separators.
+    #          Quoted separator also may be an empty string (no separator), but this is
+    #          a meaningless feature.
+    if args.F is not None:
+        df.separator = args.F
+
+    return df
+
+
+def format_str_adjust(format_str=None, data_fmt=None):
+    '''
+    Performs the following changes to the format string 'format_str':
+        1. Replaces all occurrances of '{response.data}' with '{response_data}'
+        2. Inserts separator between fields if a separator is specified
+    '''
+
+    if format_str is None:
+        return None
+
+    if data_fmt is None:
+        data_fmt = _DataPrintFormat()  # No separator will be inserted
+
+    # In 'format_str': replace all instances of '{response.data}' with '{response_data}'
+    p = re.compile("{response.data}")
+    format_str = p.sub("{response_data}", format_str)
+
+    # If a separator is specified (argument -F), then put the separators between each field
+    if data_fmt.separator is not None:
+        p = re.compile("} *{")
+        format_str = p.sub("}" + "{}".format(data_fmt.separator) + "{", format_str)
+
+    return format_str
+
+
+def print_response_data(data=None, data_fmt=None):
```

Comment:
Minor quibble: Reading the code where this is used, the name of this function momentarily misled me into thinking that it printed. Could this be named `format_response_data`?

Reply:
You are right. I renamed the function  to `format_response_data`.